### PR TITLE
SmartPlaces.Facilities documentation

### PR DESCRIPTION
**Pull request by hammar**
This PR contains StyleCop-compliant documentation for the large majority of the code (a few unclear methods remain to discuss, but these are not blockers), and rewritten README files aiming to explain the design and utility of the solutions.

### CODE REVIEW

**Comment by hammar** [comment]
So, it turns out those remaining comments actually are blockers, as StyleCop will report build errors until they are resolved, preventing GitHub CI from clearing this PR for merge. Let's discuss those comments once everyone is back from Thanksgiving. It's likely a rather simple discussion.

File: SmartPlaces.Facilities/lib/IngestionManager.Mapped/README.md [hunk 1]
```diff
@@ -1,12 +1,53 @@
...
+    var logger = sp.GetRequiredService<ILogger<MappedOntologyMappingLoader>>();
+    return new MappedOntologyMappingLoader(logger, hostContext.Configuration["ontologyMappingFilename"]);
+});
+
+services.AddSingleton<IOntologyMappingManager, OntologyMappingManager>();
+```
+
+## Usage
+
+The entry point to using these libraries is the base class `IngestionProcessorBase.IngestFromApiAsync()` method; that method in turn calls the `MappedGraphIngestionProcessor.GetSites()` method, which initiates ingestion of all sites (e.g., campuses/buildings or other suitable starting nodes) from the input graph.
```
Comment (Tyler-Angell) [inline SmartPlaces.Facilities/lib/IngestionManager.Mapped/README.md:41]: Just calling out that this GetSites could fall out of sync with your other PR

File: SmartPlaces.Facilities/lib/IngestionManager/src/AzureDigitalTwins/AzureDigitalTwinsGraphManager.cs [hunk 2]
```diff
@@ -155,6 +192,12 @@ private async Task CommitCache(List<Task> cacheTasks)
             Logger.LogInformation("Completing cache commit.");
         }
 
+        /// <summary>
+        /// Asynchronously import a set of relationships into the target ADT graph.
+        /// </summary>
+        /// <param name="relationships">Relationships to upload (map key is relationship ID).</param>
+        /// <param name="retryAttempt">Retry counter.</param>
+        /// <returns>An awaitable task.</returns>
```
Comment (Tyler-Angell) [inline SmartPlaces.Facilities/lib/IngestionManager/src/AzureDigitalTwins/AzureDigitalTwinsGraphManager.cs:200]: Any reason this method is redefining its documentation whereas `GetModedlAsync` and `UploadGraphAsync` are using `\\\<inheritdoc/>`

Comment (Tyler-Angell) [reply SmartPlaces.Facilities/lib/IngestionManager/src/AzureDigitalTwins/AzureDigitalTwinsGraphManager.cs:200]: Same question for `ImportTwinAsync`

Comment (hammar) [reply SmartPlaces.Facilities/lib/IngestionManager/src/AzureDigitalTwins/AzureDigitalTwinsGraphManager.cs:200]: Those other methods are defined in IOutputGraphManager and are implemented here; so they have inheritable docs that can be reused, whereas this method and ImportTwinAsync are only defined here, and thus their docs cannot be inherited.

File: SmartPlaces.Facilities/lib/IngestionManager.Mapped/README.md (hunk 1 above)
Comment (hammar) [reply SmartPlaces.Facilities/lib/IngestionManager.Mapped/README.md:41]: Good call-out, once the other refactoring PRs are merged I'll update this one accordingly.

File: SmartPlaces.Facilities/lib/IngestionManager.Mapped/README.md [hunk 3]
```diff
@@ -1,12 +1,53 @@
...
+
+`GetSites()` calls out to an input graph manager to actually query the source graph for those starting nodes, and iterates over child nodes returned by said graph manager. As wired up above, that input graph manager will be our own `MappedGraphManager`, which knows how to talk to the Mapped API.
```
[the remark this answers is not in the document]
Comment (hammar) [reply SmartPlaces.Facilities/lib/IngestionManager.Mapped/README.md:43]: Good call-out, once the other refactoring PRs are merged I'll update this one accordingly.

File: SmartPlaces.Facilities/lib/IngestionManager/README.md [hunk 4]
```diff
@@ -1,256 +1,69 @@
...
-*Description*
-Methods for ingesting a graph from a source graph and inserting into a target graph
+* A subclass of `IngestionProcessorBase` (implementing `IGraphIngestionProcessor`)
+* An implementation of `IInputGraphManager`
 
-**Method: IngestionFromApiAsync**
+The `IngestionProcessorBase` subclass must provide an implementation of the abstract method `GetSites()`. That method should initiate ingestion of all sites (e.g., campuses/buildings or other suitable starting nodes) from the input graph, traverse them and their child nodes (buildings, floors, rooms, etc), translating them as necessary, and storing them in the output ADT graph. 
 
-*Description*
+In order to aid in this process, the `IngestionProcessorBase` class references implementations of `IInputGraphManager` and `IOutputGraphManager` that you may use to interact with the input and output graphs, respectively. An ADT-specific implementation of the latter is provided for you in `AzureDigitalTwinsGraphManager`, but you will need to create the former, adapted to whatever data source you are ingesting from.
```
Comment (joebeernink) [inline SmartPlaces.Facilities/lib/IngestionManager/README.md:14]: Expand ADT to Azure Digital Twins

File: SmartPlaces.Facilities/lib/IngestionManager/README.md [hunk 5]
```diff
@@ -1,256 +1,69 @@
...
-
-*Description*
-
-Abstract Base class for loading a site graph from input source to output target. Implements IGraphIngestionProcessor
+    // Ingestion Manager
+    options.AzureDigitalTwinsEndpoint = hostContext.Configuration["AzureDigitalTwinsEndpoint"];
+});
+```
 
+If all is configured correctly, your code referring to `IGraphIngestionProcessor` should inject your custom `IngestionProcessorBase` subclass, on which you can call the inherited method `IngestFromApiAsync()`. That method will initialise the base processor, and then immediately call your customized `GetSites()` method, kicking off the rest of the ingestion process.
```
Comment (joebeernink) [inline SmartPlaces.Facilities/lib/IngestionManager/README.md:69]: ProcessSites

File: SmartPlaces.Facilities/lib/IngestionManager/src/AzureDigitalTwins/AzureDigitalTwinsGraphManager.cs [hunk 6]
```diff
@@ -45,20 +58,33 @@ public class AzureDigitalTwinsGraphManager<TOptions> : IOutputGraphManager
...
+        /// </summary>
         protected TelemetryClient TelemetryClient { get; }
 
+        /// <summary>
+        /// Gets twin mapping index cache.
+        /// </summary>
         protected ITwinMappingIndexer TwinMappingIndexer { get; }
 
+        /// <summary>
+        /// Gets a value indicating whether to upload graph to ADT (else, only cache will be updated).
```
Comment (joebeernink) [inline SmartPlaces.Facilities/lib/IngestionManager/src/AzureDigitalTwins/AzureDigitalTwinsGraphManager.cs:83]: Expand ADT

**Review by joebeernink — APPROVED** [review]

**Pull request merged**